ppa/archive/management/commands/hathi_images.py

```diff
@@ -0,0 +1,165 @@
+import requests
+from pathlib import Path
+from time import sleep
+
+import progressbar
+from django.core.management.base import BaseCommand, CommandError
+from django.template.defaultfilters import pluralize
+from corppa.utils.path_utils import encode_htid, get_vol_dir
+
+from ppa.archive.models import DigitizedWork
+from ppa.archive.templatetags.ppa_tags import page_image_url
+
+
+class Command(BaseCommand):
+    """
+    Download HathiTrust page image data via image server
+
+    Note: Excerpts cannot be specified individually, only by source (collectively)
+    """
+    help = __doc__
+    #: normal verbosity level
+    v_normal = 1
+    verbosity = v_normal
+    #: crawl delay (in seconds)
+    crawl_delay=1
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "out",
+            type=Path,
+            help="Top-level output directory")
+        parser.add_argument(
+            "--htids",
+            nargs="*",
+            help="Optional list HathiTrust ids to download",
+        )
+        parser.add_argument(
+            "--progress",
+            action="store_true",
+            help="Display progress bars to track download progress",
+            default=True,
+        )
+    
+    def download_image(self, page_url: str, out_file: Path) -> None:
+        response = requests.get(page_url)
```

Comment:
I would be interested in seeing debug logging / verbose output on the crawl tokens / throttling, so we can get a better sense of how that works. 

How did you come up with the crawl delay?

Reply:
HathiTrust's robots